stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -583,42 +597,105 @@ function complete_any_methods(ex_org::Expr, callee_module::Module, context_modul
     return out
 end
 
-function complete_methods_args(funargs::Vector{Any}, ex_org::Expr, context_module::Module, default_any::Bool, allow_broadcasting::Bool)
+
+function detect_invalid_kwarg!(kwargs_ex, @nospecialize(x), kwargs_flag, possible_splat)
+    n = isexpr(x, :kw) ? x.args[1] : x
+    if n isa Symbol
+        push!(kwargs_ex, n)
+        return kwargs_flag
+    end
+    possible_splat && isexpr(x, :...) && return kwargs_flag
+    return 2 # The kwarg is invalid
+end
+
+function detect_args_kwargs(funargs::Vector{Any}, context_module::Module, default_any::Bool, ::Val{broadcasting}) where {broadcasting}
```

Comment:
This never should be a `Val`. That just makes the API smelly and performance bad for no reason.
```suggestion
function detect_args_kwargs(funargs::Vector{Any}, context_module::Module, default_any::Bool, broadcasting::Bool)
```